dnf.spec

```diff
@@ -1,44 +1,78 @@
+# default dependencies
 %global hawkey_version 0.13.3
 %global librepo_version 1.9.0
 %global libcomps_version 0.1.8
 %global rpm_version 4.14.0
+
+# conflicts
+%global conflicts_dnf_plugins_core_version 2.1.3
+
+# override dependencies for rhel 7
 %if 0%{?rhel} == 7
-%global rpm_version 4.11.3-27
+    %global rpm_version 4.11.3-25.el7.centos.1
 %endif
+
+# override dependencies for fedora 26
 %if 0%{?fedora} == 26
-%global rpm_version 4.13.0.1-7
+    %global rpm_version 4.13.0.1-7
+%endif
+
+
+# default build options
+# %%bcond_without sets an --without option -> %%{with foo} is true
+# overriding with %%undefine is not ideal, but it avoids complicated nested conditions
+%bcond_without python2
+%bcond_without python3
+
+# override build options for rhel <= 7
+%if 0%{?rhel} && 0%{?rhel} <= 7
+    %undefine with_python3
 %endif
-%if 0%{?fedora} > 26
-%global rpm_version 4.13.90
+
+# override build options for rhel >= 8
+%if 0%{?rhel} >= 8
+    %undefine with_python2
 %endif
-%global min_plugins_core 2.1.3
-%global dnf_langpacks_ver 0.15.1-6
 
-%global confdir %{_sysconfdir}/%{name}
 
-%global pluginconfpath %{confdir}/plugins
-%global py2pluginpath %{python2_sitelib}/%{name}-plugins
+# configurable name for the compat yum package
+%global yum_subpackage_name %{name}-yum
 
+# provide yum4 on rhel <= 7 to avoid conflict with existing yum
 %if 0%{?rhel} && 0%{?rhel} <= 7
-%bcond_with python3
-%else
-%bcond_without python3
+    %global yum_subpackage_name yum4
+%endif
+
+# provide yum on rhel >= 8, it replaces old yum
+%if 0%{?rhel} && 0%{?rhel} >= 8
+    %global yum_subpackage_name yum
+%endif
+
+
+# paths
+%global confdir %{_sysconfdir}/%{name}
+%global pluginconfpath %{confdir}/plugins
+
+%if %{with python2}
+    %global py2pluginpath %{python2_sitelib}/%{name}-plugins
 %endif
 
 %if %{with python3}
-%global py3pluginpath %{python3_sitelib}/%{name}-plugins
+    %global py3pluginpath %{python3_sitelib}/%{name}-plugins
 %endif
 
 # Use the same directory of the main package for subpackage licence and docs
 %global _docdir_fmt %{name}
 
-%global yum_package_description \
-As a Yum CLI compatibility layer, supplies /usr/bin/yum redirecting to DNF.
+
+%global pkg_summary     Package manager
```

Comment:
this is very pointless description

Reply:
very pointless comment (sorry, couldn't resist)
What do you propose?

Reply:
Even original summary was better